src/formatDate.js

```diff
@@ -1,14 +1,34 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      receivedDate['YY'] = dateArr[i].slice(2);
+    }
+
+    if (fromFormat[i] === 'YY') {
+      receivedDate['YYYY'] = (dateArr[i] < 30 ? 20 : 19) + dateArr[i];
```

Comment:
When converting from 'YY' to 'YYYY', you're using string concatenation which can lead to incorrect results if `dateArr[i]` is not explicitly converted to a string. Ensure that you convert the year to a string before concatenating it to '20' or '19'.